src/Umbraco.Web.BackOffice/Filters/UserGroupValidateAttribute.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Net;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Umbraco.Core;
+using Umbraco.Core.Mapping;
+using Umbraco.Core.Models.Membership;
+using Umbraco.Core.Services;
+using Umbraco.Web.Common.ActionResults;
+using Umbraco.Web.Common.Exceptions;
+using Umbraco.Web.Models.ContentEditing;
+
+namespace Umbraco.Web.BackOffice.Filters
+{
+    internal sealed class UserGroupValidateAttribute : TypeFilterAttribute
+    {
+        public UserGroupValidateAttribute() : base(typeof(UserGroupValidateFilter))
+        {
+        }
+
+        private class UserGroupValidateFilter : IActionFilter
+        {
+              private readonly IUserService _userService;
+              private readonly UmbracoMapper _umbracoMapper;
+
+              public UserGroupValidateFilter(
+            IUserService userService,
+            UmbracoMapper umbracoMapper)
+        {
+            _userService = userService ?? throw new ArgumentNullException(nameof(userService));
+            _umbracoMapper = umbracoMapper ?? throw new ArgumentNullException(nameof(umbracoMapper));
+        }
+
+        public void OnActionExecuting(ActionExecutingContext context)
+        {
+            var userGroupSave = (UserGroupSave) context.ActionArguments["userGroupSave"];
+
+            userGroupSave.Name = userGroupSave.Name.CleanForXss('[', ']', '(', ')', ':');
+            userGroupSave.Alias = userGroupSave.Alias.CleanForXss('[', ']', '(', ')', ':');
+
+            //Validate the usergroup exists or create one if required
+            IUserGroup persisted;
+            switch (userGroupSave.Action)
+            {
+                case ContentSaveAction.Save:
+                    persisted = _userService.GetUserGroupById(Convert.ToInt32(userGroupSave.Id));
+                    if (persisted == null)
+                    {
+                        var message = $"User group with id: {userGroupSave.Id} was not found";
+                        context.Result = new UmbracoErrorResult(HttpStatusCode.NotFound, message);
+                        return;
+                    }
+
+                    if (persisted.Alias != userGroupSave.Alias && persisted.IsSystemUserGroup())
+                    {
+                        var message = $"User group with alias: {persisted.Alias} cannot be changed";
+                        context.Result = new UmbracoErrorResult(HttpStatusCode.BadRequest, message);
+                        return;
+                    }
+
+                    //map the model to the persisted instance
```

Comment:
These comments here and just below are more what rather than why

Reply:
Agree.. But again, these are just copied over one to one